src/mpid/ch4/shm/ipc/xpmem/xpmem_pre.h

```diff
@@ -8,17 +8,20 @@
 
 #ifdef MPIDI_CH4_SHM_ENABLE_XPMEM
 #include <xpmem.h>
+#include "xpmem_types.h"
```

Comment:
`xpmem_types.h` is internal to xpmem.  It should not be added to the pre header, which is being exposed to the upper layer.  In fact, `xpmem.h` should not be included either.